cycler.py

```diff
@@ -38,6 +38,30 @@
     {'color': 'b', 'linestyle': '-'}
     {'color': 'b', 'linestyle': '--'}
     {'color': 'b', 'linestyle': '-.'}
+
+RememberTheStyle
+================
+
+Helper class for mapping keys -> styles::
+
+   from cycler import RememberTheStyle, cycler
+
+   rts = RememberTheStyle((cycler('c', 'rgb') +
+                           cycler('ls', ['-', '--', ':']))
+
+   rts['cat']
```

Comment:
I find the overridden ``__getitem__`` really non-intuitive and would highly recommend using a method for this purpose. Something like:

```
class NamedCycler:
    def __init__(self, cycler, names=None):
        self.cycler = cycler
        if names:
            self._cache = dict(zip(names, cycler))
        else:
            self._cache = {}

    def associated_name(self, key):
        """
        Return a cycler item for the given key. If the key hasn't already been seen, the
        item will come from the next iteration of the cycler.

        """
        if key not in self._cache:
            self._cache[key] = next(self.cycler)
        return self._cache[key]
```